teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,473 @@
+last-updated: 2020-04-23
+status: proposed
+---
+
+# Tekton Enhancement Proposal Process
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Stewardship](#stewardship)
+- [Reference-level explanation](#reference-level-explanation)
+  - [What type of work should be tracked by a TEP](#what-type-of-work-should-be-tracked-by-a-tep)
+  - [TEP Template](#tep-template)
+  - [TEP Metadata](#tep-metadata)
+  - [TEP Workflow](#tep-workflow)
+  - [Git Implementation](#git-implementation)
+  - [Prior Art](#prior-art)
+- [Examples](#examples)
+  - [Share Task and Pipeline as OCI artifact](#share-task-and-pipeline-as-oci-artifact)
+  - [PipelineResource re-design](#pipelineresource-re-design)
+- [Drawbacks](#drawbacks)
+- [Unresolved Questions](#unresolved-questions)
+<!-- /toc -->
+
+
+## Summary
+
+A standardized development process for Tekton is proposed in order to
+
+- provide a common structure and clear checkpoints for proposing
+  changes to Tekton
+- ensure that the motivation for a change is clear
+- allow for the enumeration stability milestones and stability
+  graduation criteria
+- persist project information in a Version Control System (VCS) for
+  future Tekton users and contributors
+- support the creation of _high value user facing_ information such
+  as:
+  - an overall project development roadmap
+  - motivation for impactful user facing changes
+- ensure community participants are successfully able to drive changes
+  to completion across one or more releases while stakeholders are
+  adequately represented throughout the process
+
+This process is supported by a unit of work called a Tekton
+Enhancement Proposal (TEP). A TEP attempts to combine aspects of the
+following:
+
+- feature, and effort tracking document
+- a product requirements document
+- design document
+
+into one file which is created incrementally in collaboration with one
+or more [Working
+Groups](https://github.com/tektoncd/community/blob/master/working-groups.md)
+(WGs).
+
+This process does not block authors from doing early design docs using
+any means. It does not block authors from sharing those design docs
+with the community (during Working groups, on Slack, GitHub, …).
+
+**This process acts as a requirement when a design docs is ready to be
+implemented or integrated in the `tektoncd` projects**. In other words,
+a change that impacts other `tektoncd` projects or users cannot be
+merged if there is no `TEP` associated with it. Bug fixes and small
+changes like refactoring that do not affect the APIs (CRDs, REST APIs)
+are not concerned by this. Fixing the behaviour of a malfunctioning
+part of the project is also not concerned by this.
+
+This TEP process is related to
+- the generation of an architectural roadmap
+- the fact that the what constitutes a feature is still undefined
+- issue management
+- the difference between an accepted design and a proposal
+- the organization of design proposals
```

Comment:
this is a total pet peeve but a bunch of these bullet point lists copied from k8s don't have the first letter capitalized and it looks weird. i totally understand if you want to ignore this feedback but I think the lists look better with the first letter capitalized.

Reply:
interesting :stuck_out_tongue_closed_eyes: I tend to think the opposite, or more accurately, if I don't see a dot — ending the sentence — I have trouble capitalizing :angel:

Reply:
hah!

now that i think about it i think you might be right - i took a class once (where no one could ever get more than 85% on anything, another story) and i think the way you are capitalizing might be better 🤔 

i was hoping https://github.com/tektoncd/website/blob/master/content/en/doc-con-formatting.md would help but i guess that's too crazy specific XD maybe @sergetron can tell us what's right

Reply:
The whole point of lists is that they're quickly scannable, and for that, they must be homogeneous.

The TL;DR is: keep the list item structure and grammar as homogeneous as possible, capitalize the first word of each item, and punctuate the ends as follows: 

* If most of your list items are single words or 2-3 word phrases that do not contain verbs, but one or two are not, rewrite the non-conforming ones to fit the rest and do not put periods at the end.

* If most of your list items contain verbs and/or are complete sentences, rewrite the non-conforming ones to fit the rest and then put a period at the end of each list.

More info here: https://developers.google.com/style/lists

Reply:
PR: https://github.com/tektoncd/website/pull/139

Reply:
bumping this, but not enough to block merging on!